optics_functions/coupling.py

```diff
@@ -0,0 +1,208 @@
+"""
+Coupling
+********
+
+Functions to calculate coupling from twiss dataframes.
+
+"""
+import logging
+from contextlib import suppress
+from typing import Sequence
+
+import numpy as np
+from tfs import TfsDataFrame
+
+from optics_functions.constants import (ALPHA, BETA, GAMMA,
+                                        X, Y, TUNE, DELTA,
+                                        MINIMUM, PI2, PHASE_ADV, S, LENGTH)
+from optics_functions.rdt import calculate_rdts
+from optics_functions.utils import split_complex_columns, timeit
+
+COUPLING_RDTS = ['F1001', 'F1010']
+
+
+LOG = logging.getLogger(__name__)
+
+
+def coupling_from_rdts(df: TfsDataFrame, real: bool = False, **kwargs):
+    """ Returns the coupling term.
+
+    .. warning::
+        This function changes sign of the real part of the RDTs compared to
+        [FranchiAnalyticFormulas2017]_ to be consistent with the RDT
+        calculations from [CalagaBetatronCoupling2005]_ .
+
+    Args:
+        df (TfsDataFrame): Twiss Dataframe
+        real (bool): Split complex columns into two real-valued columns.
```

Comment:
Havinig `real` as argument makes it sound like it'd only return the real part. Maybe something like `split_complex` could be more explicit?

Reply:
that sounds a bit ambiguous. As if we would only split, if something is complex.
How about calling the flag "complex" and inverting?

Reply:
Sounds good!